podcasts/Bookmarks/Editing/BookmarkEditTitleView.swift

```diff
@@ -0,0 +1,205 @@
+                }
+        }
+        .frame(maxWidth: .infinity)
+        .padding()
+        .background(theme.playerBackground01)
+        .onChange(of: viewModel.didAppear) { _ in
+            focusedField = .title
+        }
+    }
+
+    // MARK: - Views
+
+    /// The actual content of the view
+    private var mainView: some View {
+        VStack(spacing: EditConstants.padding) {
+            headerView
+            Spacer()
+            textField
+            Spacer()
+            saveButton
+        }
+        .padding(.top, EditConstants.padding)
+    }
+
+    /// The title and subtitle views
+    @ViewBuilder
+    private var headerView: some View {
+        Text(headerTitle)
+            .foregroundStyle(theme.playerContrast01)
+            .font(size: 19, style: .title3, weight: .bold)
+
+        Text(headerSubTitle)
+            .foregroundStyle(theme.playerContrast02)
+            .font(style: .callout)
+    }
+
+    /// A line that appears under the text field
+    @ViewBuilder
+    private var textFieldUnderline: some View {
+        VStack {
+            Spacer()
+
+            Divider().background(theme.playerContrast05)
+        }
+        .offset(y: 6)
+    }
+
+    @ViewBuilder
+    private var saveButton: some View {
+        let background = theme.playerHighlight01
+
+        // Calculate the text color to make sure the button text is readable
+        let text: Color = background.luminance() < 0.5 ? .white : .black
+
+        Button(L10n.saveBookmark) {
+            viewModel.save(title: bookmarkTitle)
+        }
+        .buttonStyle(BasicButtonStyle(textColor: text, backgroundColor: background))
+    }
+
+
+    @ViewBuilder
+    private var textField: some View {
+        let prompt = Text(placeholder).foregroundColor(theme.playerContrast05)
+
+        ZStack {
+            /// This looks really bad and I bet you may have questions...
+            ///
+            /// So, there's this _really_ fun bug that is causing the TextField's height to bounce between the max size
+            /// and the scaled size, whicn causes the entire view to jump. Believe me I tried everything I could.
+            ///
+            /// So! to fix this I've crafted this monstrosity of code below. So here's what it does:
+            ///
+            /// - It creates an invisible Text view and a TextField with the same font, and min scaling
+            /// - When the TextField changes the Text is also updated and will scale the same way the text field should
+            /// - Since the Text view doesn't have the same bug we listen for content size changes
+            /// - Then we sync the Text's height to the TextField to prevent it from jumping around.
+            ///
+            /// It works 🤷‍♀️
+            ///
+            /// Feel free to comment out the `.frame(height: textFieldSize.height)` to see the bug in action.
+            ///
+            ContentSizeReader(contentSize: $textFieldSize) {
```

Comment:
Please let me know if there is a better way to solve this. I tried really hard, but this is what I got.